pkg/agent/controller/agent.go

```diff
@@ -486,18 +499,10 @@ func (a *Controller) newServiceImport(svcExport *mcsv1a1.ServiceExport) *mcsv1a1
 
 func (a *Controller) getIPsAndPortsForService(service *corev1.Service, siType mcsv1a1.ServiceImportType) (
 	[]string, []mcsv1a1.ServicePort, error) {
-	var mcsPorts = make([]mcsv1a1.ServicePort, 0, len(service.Spec.Ports))
-
-	for _, port := range service.Spec.Ports {
-		mcsPorts = append(mcsPorts, mcsv1a1.ServicePort{
-			Name:     port.Name,
-			Protocol: port.Protocol,
-			Port:     port.Port,
-		})
-	}
+	mcsPorts := a.getPortsForService(service)
 
 	if siType == mcsv1a1.ClusterSetIP {
-		mcsIP := getGlobalIPFromService(service)
+		mcsIP, _, _ := a.getGlobalIP(service)
```

Comment:
We no longer enter this path with GN enabled so we don't need to call `getGlobalIP`.